sqliterepo/repository.c

```diff
@@ -1617,34 +1617,51 @@ sqlx_repository_dump_base_fd_no_copy(struct sqlx_sqlite3_s *sq3,
 	EXTRA_ASSERT(read_file_cb != NULL);
 	const char *fallback_msg = "falling back on legacy dump function";
 
+	/* Flush cache */
 	if ((rc = sqlite3_db_cacheflush(sq3->db)) != SQLITE_OK) {
 		GRID_NOTICE("Failed to flush [%s][%s]: %s, %s",
 					sq3->name.base, sq3->name.type, sqlite_strerror(rc),
 					fallback_msg);
-		err = sqlx_repository_dump_base_fd(sq3, read_file_cb, cb_arg);
```

Comment:
I looks like this statement is gone. This means that we may serve a non flushed database.
I suggest running the integrity check before the cache flush.